CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift

```diff
@@ -295,35 +295,6 @@ public let ATTRIBUTE_NODES: [Node] = [
     ]
   ),
 
-  Node(
-    kind: .declName,
-    base: .syntax,
-    nameForDiagnostics: "declaration name",
-    children: [
-      Child(
-        name: "BaseName",
-        deprecatedName: "DeclBaseName",
-        kind: .token(choices: [
-          .token(tokenKind: "IdentifierToken"),
-          .token(tokenKind: "BinaryOperatorToken"),
-          .keyword(text: "init"),
-          .keyword(text: "self"),
-          .keyword(text: "Self"),
-        ]),
-        nameForDiagnostics: "base name",
-        documentation: "The base name of the protocol's requirement."
-      ),
-      Child(
-        name: "Arguments",
-        deprecatedName: "DeclNameArguments",
```

Comment:
`IdentifierExpr` uses `DeclNameArguments` as its child name still. Or was that updated in the other PR and just not merged yet?

Also it's a bit weird that `dynamicReplacementAttributeArguments` now contains an expression.

And `QualifiedDeclNameSyntax` also has a `name`/`arguments` split vs just having a ... well, now `IdentifierExpr` (but I still feel that's a bit weird).

Reply:
> `IdentifierExpr` uses `DeclNameArguments` as its child name still. Or was that updated in the other PR and just not merged yet?

That’s the whole point. `IdentifierExpr` should be the main user of `DeclNameArguments`.

> Also it's a bit weird that `dynamicReplacementAttributeArguments` now contains an expression.

I guess it does make some sense. You are specifying the decl that it should replace using an expression just like you reference a declaration using an expression in normal source code.

> And `QualifiedDeclNameSyntax` also has a `name`/`arguments` split vs just having a ... well, now `IdentifierExpr` (but I still feel that's a bit weird).

`QualifiedDeclNameSyntax` is only used for differentiability attributes. I’d like to investigate if we can remove it altogether but that’s something for another PR.

Reply:
> That’s the whole point. IdentifierExpr should be the main user of DeclNameArguments.

I meant its child is named "DeclNameArguments" rather than "Arguments"

> QualifiedDeclNameSyntax is only used for differentiability attributes. I’d like to investigate if we can remove it altogether but that’s something for another PR.

Sure, that's fine 👍

Reply:
I prefer `DeclNameArguments`. `Arguments` suggests too much that `IdentifierExpr` is performing a function call.

Reply:
`MemberAccessExprSyntax` is yet another 😅